pkg/scheduler/reconciliation/reconciliation.go

```diff
@@ -20,6 +20,7 @@ const (
 type Repository interface {
 	CreateReconciliation(state *cluster.State, cfg *model.ReconciliationSequenceConfig) (*model.ReconciliationEntity, error)
 	RemoveReconciliation(schedulingID string) error
+	RemoveReconciliations(reconEntities []*model.ReconciliationEntity) error
```

Comment:
I think this interface method is requiring "too much" - you have to load entire object into memory just to delete it. It would be much better to only require a slice of schedulingIDs (strings).